### PR TITLE
feat: surface auth form field labels for i18n

### PR DESCRIPTION
**Pull request by nadilas**
Surface auth form field labels for i18n on:
- magic-link-form
- password-form

refactor field labels to be uniformly named:
- update-password-form
- forgot-password-form

### CODE REVIEW

File: packages/saas-ui-auth/src/components/forgot-password-form.tsx [hunk 1]
```diff
@@ -38,7 +38,7 @@ export const ForgotPasswordForm: React.FC<ForgotPasswordFormProps> = ({
   onError = () => null,
   onValidationError,
   submitLabel,
-  label,
```
Comment (Pagebakers) [inline packages/saas-ui-auth/src/components/forgot-password-form.tsx:41]: Good idea to rename these! But let's keep them in here, as I don't want to introduce breaking changes in a minor update.

Add a comment to deprecate them and fallback to label if emailLabel isn't set.
```
/**
 * @deprecated
 * /
 label?: string
```

**Review by Pagebakers — CHANGES_REQUESTED** [review]
Almost good to go.

File: packages/saas-ui-auth/src/components/forgot-password-form.tsx (hunk 1 above)
Comment (Pagebakers) [reply packages/saas-ui-auth/src/components/forgot-password-form.tsx:41]: Applies to all labels that have been renamed 🙏

Comment (nadilas) [reply packages/saas-ui-auth/src/components/forgot-password-form.tsx:41]: It was implied 👍 please check which fallback order you prefer `new ?? old` vs `old ?? new`

**Review by Pagebakers — APPROVED** [review]
LTM

**Pull request merged**